kernel/src/engine/parquet_row_group_skipping/tests.rs

```diff
@@ -0,0 +1,439 @@
+            &DataType::DOUBLE
+        ),
+        Some(1147f64.into())
+    );
+
+    // type widening!
+    assert_eq!(
+        filter.get_min_stat_value(
+            &col_name_to_path("numeric.floats.float32"),
+            &DataType::DOUBLE
+        ),
+        Some(139f64.into())
+    );
+
+    assert_eq!(
+        filter.get_min_stat_value(
+            &col_name_to_path("numeric.floats.float32"),
+            &DataType::FLOAT
+        ),
+        Some(139f32.into())
+    );
+
+    assert_eq!(
+        filter.get_min_stat_value(&col_name_to_path("bool"), &DataType::BOOLEAN),
+        Some(false.into())
+    );
+
+    assert_eq!(
+        filter.get_min_stat_value(&col_name_to_path("varlen.binary"), &DataType::BINARY),
+        Some([].as_slice().into())
+    );
+
+    // CHEAT: Interpret the decimal128 column's fixed-len array as binary
+    assert_eq!(
+        filter.get_min_stat_value(
+            &col_name_to_path("numeric.decimals.decimal128"),
+            &DataType::BINARY
+        ),
+        Some(
+            [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0x2b, 0x78]
+                .as_slice()
+                .into()
+        )
+    );
+
+    assert_eq!(
+        filter.get_min_stat_value(
+            &col_name_to_path("numeric.decimals.decimal32"),
+            &DataType::decimal(8, 3).unwrap()
+        ),
+        Some(Scalar::Decimal(11032, 8, 3))
+    );
+
+    assert_eq!(
+        filter.get_min_stat_value(
+            &col_name_to_path("numeric.decimals.decimal64"),
+            &DataType::decimal(16, 3).unwrap()
+        ),
+        Some(Scalar::Decimal(11064, 16, 3))
+    );
+
+    // type widening!
+    assert_eq!(
+        filter.get_min_stat_value(
+            &col_name_to_path("numeric.decimals.decimal32"),
+            &DataType::decimal(16, 3).unwrap()
+        ),
+        Some(Scalar::Decimal(11032, 16, 3))
+    );
+
+    assert_eq!(
+        filter.get_min_stat_value(
+            &col_name_to_path("numeric.decimals.decimal128"),
+            &DataType::decimal(32, 3).unwrap()
+        ),
+        Some(Scalar::Decimal(11128, 32, 3))
+    );
+
+    // type widening!
+    assert_eq!(
+        filter.get_min_stat_value(
+            &col_name_to_path("numeric.decimals.decimal64"),
+            &DataType::decimal(32, 3).unwrap()
+        ),
+        Some(Scalar::Decimal(11064, 32, 3))
+    );
+
+    // type widening!
+    assert_eq!(
+        filter.get_min_stat_value(
+            &col_name_to_path("numeric.decimals.decimal32"),
+            &DataType::decimal(32, 3).unwrap()
+        ),
+        Some(Scalar::Decimal(11032, 32, 3))
+    );
+
+    assert_eq!(
+        filter.get_min_stat_value(&col_name_to_path("chrono.date32"), &DataType::DATE),
+        Some(PrimitiveType::Date.parse_scalar("1971-01-01").unwrap())
+    );
+
+    assert_eq!(
+        filter.get_min_stat_value(&col_name_to_path("chrono.timestamp"), &DataType::TIMESTAMP),
+        None // Timestamp defaults to 96-bit, which doesn't get stats
```

Comment:
int96 just never writes stats? that's common writer pattern or a made up case here?

Reply:
I mean, it's _allowed_ to return stats, but parquet-mr and other common writers don't do so because the type was deprecated about 6 years ago. In particular, the parquet file I created using pyspark does not have stats on the int96 column:
```
Row group 0:  count: 5  181.00 B records  start: 4  total(compressed): 905 B total(uncompressed):940 B 
--------------------------------------------------------------------------------
                             type      encodings count     avg size   nulls   min / max
bool                         BOOLEAN   S   _     5         6.40 B     3       "false" / "true"
chrono.date32                INT32     S   _     5         10.40 B    0       "1971-01-01" / "1971-01-05"
chrono.timestamp             INT96     S _ R     5         21.20 B    0       
chrono.timestamp_ntz         INT64     S   _     5         13.60 B    0       "1970-01-02T00:00:00.000000" / "1970-01-02T00:04:00.000000"
numeric.decimals.decimal128  FIXED[14] S   _     5         14.80 B  0       "11.128" / "15.128"
numeric.decimals.decimal32   INT32     S   _     5         10.40 B    0       "11.032" / "15.032"
numeric.decimals.decimal64   INT64     S   _     5         13.00 B    0       "11.064" / "15.064"
numeric.floats.float32       FLOAT     S   _     5         10.40 B    0       "139.0" / "1048699.0"
numeric.floats.float64       DOUBLE    S   _     5         14.20 B    0       "1147.0" / "1.125899906842747E15"
numeric.ints.int16           INT32     S   _     5         10.40 B    0       "1000" / "1004"
numeric.ints.int32           INT32     S   _     5         10.40 B    0       "1000000" / "1000004"
numeric.ints.int64           INT64     S   _     5         12.80 B    0       "1000000000" / "1000000004"
numeric.ints.int8            INT32     S   _     5         10.20 B    0       "0" / "4"
varlen.binary                BINARY    S   _     5         11.60 B    0       "0x" / "0x00000000"
varlen.utf8                  BINARY    S   _     5         11.20 B    0       "a" / "e"
```
(the same parquet footer information is pasted as a doc comment in the unit tests)

Reply:
cool thanks for the context :)